### ISSUE TITLE
Generate `QNProblemSet`s without the `StateTransitionManager` 

### ISSUE THREAD

**Issue by grayson-helmholz** [ComPWA/qrules#284]
Continuing the recreation of the functionality of the `StateTransitionManager` of #272, the generation of the `QNProblemSet`s could also be done without it.

Today, everything between a state definition and a `QNProblemSet` is hidden in the `StateTransitionManager`: it builds the initial facts, permutates the topologies, determines the interaction settings per node, and groups the resulting `ProblemSet`s by interaction strength. None of those steps is reachable on its own, so a user who wants to inspect or modify a `QNProblemSet` has to reconstruct the manager's internals.

The stages should therefore become free functions that exchange explicit data structures, so that each intermediate result can be inspected and passed on by hand. Roughly:

1. Build the initial facts and topologies for the given initial and final state.
2. Determine the solving settings for every edge and node, given the allowed interaction types (#307) and the allowed intermediate particles (#306).
3. Turn those into `ProblemSet`s and convert them to `QNProblemSet`s, grouped per interaction strength.

The default use case should still be a single call, so that the common path stays as short as the `StateTransitionManager` is now.

See also #20 and #305.

**Comment by redeboer** [comment]
Yes good idea, this is also related to https://github.com/ComPWA/qrules/issues/20.

Perhaps it's best to start writing out the steps that are required to generate `QNProblemSet`s, for instance, by summarizing how the STM does it. We could then think of an interface or set of functions that can do this.
